### DISCUSSION TITLE
Feedback / performance

### DISCUSSION THREAD

**Discussion by attet** [wezterm/wezterm#2014] [General]
Hi,

Thanks for creating such an awesome cross-platform terminal, especially the quick select and well working copy mode are killer features that really make me want to stick to this one!

I was about to rule wezterm as too laggy for my use. By laggy I mean input lag that would constantly deviate (I have my keyboard repeat rate at 40-50, so relatively high but not exceptionally high I suppose); holding down enter in shell would frequently show inconsisently skipped lines or just holding down a character or arrow key in vim the speed would visibly and annoingly fluctuate, making it hard to stop at a specific line and just ugly to look at when scrolling. This was extremely baffling given that other, cpu-hungry, terminals like tabby would scroll smooth as butter despite using ~10x the cpu (~15% vs ~150% while I'm keeping keyboard and the UI as stressed as I can).

I found out that in my case setting cpufreq governor from powersave to performance made a huge difference (this is on Debian testing; hardware  i9900k, 2800ti, 32G ram so performance capabilities per se shouldn't be an issue), getting things from 'intolerable' to 'not perfect, but given the superior features I'll allow it'.

Anyway, thanks again for creating a great terminal, and I look forward to its future developments; especially related to pane controls as I want to be able to just fire up a large window with three equally wide panes which right now takes an awful lot of manual work.

**Comment by usagi-flow** [comment]
FYI: A new opt-in feature was pushed which introduces a new rendering frontend (which in turn uses the appropriate rendering backend): https://github.com/wez/wezterm/issues/2756

Using this new feature greatly reduced CPU usage for me; it might help you too.
